### PR TITLE
feat: integrate lite-psm

### PR DESCRIPTION
**Pull request by piavgh**

### CODE REVIEW

File: pkg/liquidity-source/litepsm/constant.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+var (
+	DefaultGas = Gas{SellGem: 115000, BuyGem: 115000}
+	WAD        = number.Number_1e18
+
+	// 2^256 - 1 (type(uint256).max)
+	HALTED = new(uint256.Int).Sub(
+		new(uint256.Int).Lsh(number.Number_1, 256),
+		number.Number_1,
+	)
```
Comment (NgoKimPhu) [inline pkg/liquidity-source/litepsm/constant.go:31]: minor
```suggestion
	HALTED = new(uint256.Int).SetAllOne()
```

File: pkg/liquidity-source/litepsm/pool_simulator.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+			},
+			Fee: &pool.TokenAmount{
+				Token:  tokenOut,
+				Amount: fee.ToBig(),
+			},
+			Gas: p.gas.BuyGem,
+		}, nil
+	}
+
+	gemAmt, fee, err := p.litePSM.sellGem(amountIn)
```
Comment (NgoKimPhu) [inline pkg/liquidity-source/litepsm/pool_simulator.go:98]: do we need to verify that token in is gem?

File: pkg/liquidity-source/litepsm/pool_tracker.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
+package litepsm
+
+import (
+	"context"
+	"encoding/json"
```
Comment (NgoKimPhu) [inline pkg/liquidity-source/litepsm/pool_tracker.go:5]: (side discussion) I rmb we used to use goccy, is it not worth it?

**Review by NgoKimPhu — APPROVED** [review]

File: pkg/liquidity-source/litepsm/pool_simulator.go (hunk 2 above)
Comment (piavgh) [reply pkg/liquidity-source/litepsm/pool_simulator.go]: valid point. I added a check

File: pkg/liquidity-source/litepsm/constant.go (hunk 1 above)
Comment (piavgh) [reply pkg/liquidity-source/litepsm/constant.go:31]: thanks, I didn't know that method SetAllOne exists :D

**Pull request merged**